Ginger/Ginger/AutomatePageLib/AddActionMenu/ActionsFactory.cs

```diff
@@ -343,8 +343,9 @@ public static void AddActivitiesFromSRHandler(List<Activity> sharedActivitiesToA
                     //map activities target application to BF if missing in BF
                     userSelection = businessFlow.MapTAToBF(userSelection, activityIns, WorkSpace.Instance.Solution.ApplicationPlatforms);
                     businessFlow.SetActivityTargetApplication(activityIns);
-                    if (insertIndex >= 0)
+                    if (insertIndex >= 0 && insertIndex < businessFlow.ActivitiesGroups.Count)
                     {
+                        //QUESTION: Why do we have to move activity group?
```

Comment:
The indexing is 0-based, the clarification is needed if that piece of code is even needed in the first place. But at this point we don't want to remove the code because it might cause other bugs. And currently we are able to fix the existing bug by just adding the simple bound check.